File: backend/src/main/java/com/stepx/stepx/dto/UserDTO.java
package com.stepx.stepx.dto;

import java.util.List;

public record UserDTO(
    Long id,
    String imageString,
    String firstname,
    String lastName,
    List<String> roles,
    String username,
    String email,
    List<OrderShoesDTO> orders,
    String password
) {
}

File: frontend/src/app/dtos/user.dto.ts
import { OrderShoesDTO } from './ordershoes.dto'; 


export interface UserDTO {
    id: number;
    imageString?: string | null;
    firstname: string;
    lastName: string;
    roles: string[];
    username: string;
    email: string;
    orders?: OrderShoesDTO[] | null;
    password: string;
  }
  